### ISSUE TITLE
output type set to RAWMARYXML causes processing error

### ISSUE THREAD

**Issue by Munzey** [marytts/marytts#280]
except for input type set to TEXT, all other inputs cause the processing to fail

**Comment by psibre** [comment]
`SIMPLEPHONEMES` confirmed:
```
2015-02-03 15:27:44,240 [I/O dispatcher 1] INFO  marytts.R 7 New request (input type "SIMPLEPHONEMES", output type "AUDIO", voice "cmu-slt-hsmm", effect "", audio "WAVE")
2015-02-03 15:27:44,240 [I/O dispatcher 1] INFO  marytts.server Read: 'wEl-k@m t@ D@ 'wr=ld @v 'spitS 'sIn-T@-sIs

2015-02-03 15:27:44,241 [I/O dispatcher 1] INFO  marytts.R 7 Handling request using the following modules:
2015-02-03 15:27:44,241 [I/O dispatcher 1] INFO  marytts.R 7 - SimplePhoneme2AP (marytts.modules.SimplePhoneme2AP)
2015-02-03 15:27:44,241 [I/O dispatcher 1] INFO  marytts.R 7 - Synthesis (marytts.modules.Synthesis)
2015-02-03 15:27:44,241 [I/O dispatcher 1] INFO  marytts.R 7 Next module: SimplePhoneme2AP
2015-02-03 15:27:44,242 [I/O dispatcher 1] INFO  marytts.R 7 Next module: Synthesis
2015-02-03 15:27:44,242 [I/O dispatcher 1] INFO  marytts.HMMSynthesizer Synthesizing one sentence.
2015-02-03 15:27:44,243 [I/O dispatcher 1] ERROR marytts.server Processing failed.
java.lang.AssertionError: Unexpected tag name: expected syllable, got t
	at marytts.features.MaryGenericFeatureProcessors$SyllableNavigator.getElement(MaryGenericFeatureProcessors.java:276)
	at marytts.features.MaryGenericFeatureProcessors$Accented.process(MaryGenericFeatureProcessors.java:744)
	at marytts.features.TargetFeatureComputer.computeFeatureVector(TargetFeatureComputer.java:138)
	at marytts.modules.TargetFeatureLister.getListTargetFeatures(TargetFeatureLister.java:122)
	at marytts.modules.synthesis.HMMSynthesizer.synthesize(HMMSynthesizer.java:268)
	at marytts.modules.synthesis.Voice.synthesize(Voice.java:521)
	at marytts.modules.Synthesis.synthesizeOneSection(Synthesis.java:252)
	at marytts.modules.Synthesis.process(Synthesis.java:222)
	at marytts.server.Request.processOneChunk(Request.java:548)
	at marytts.server.Request.processOrLookupOneChunk(Request.java:385)
	at marytts.server.Request.process(Request.java:289)
	at marytts.server.http.SynthesisRequestHandler.process(SynthesisRequestHandler.java:261)
	at marytts.server.http.SynthesisRequestHandler.handleClientRequest(SynthesisRequestHandler.java:91)
	at marytts.server.http.BaseHttpRequestHandler.handle(BaseHttpRequestHandler.java:138)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler$RequestHandlerAdaptor.handle(BufferingHttpServiceHandler.java:189)
	at org.apache.http.nio.protocol.SimpleNHttpRequestHandler.handle(SimpleNHttpRequestHandler.java:51)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.processRequest(AsyncNHttpServiceHandler.java:453)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.requestReceived(AsyncNHttpServiceHandler.java:225)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler.requestReceived(BufferingHttpServiceHandler.java:127)
	at org.apache.http.impl.nio.DefaultNHttpServerConnection.consumeInput(DefaultNHttpServerConnection.java:161)
	at org.apache.http.impl.nio.DefaultServerIOEventDispatch.inputReady(DefaultServerIOEventDispatch.java:147)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:161)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:335)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:275)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:542)
	at java.lang.Thread.run(Thread.java:745)
2015-02-03 15:27:44,244 [I/O dispatcher 1] INFO  marytts.server Request couldn't be handled successfully.
```

`SABLE` confirmed:
```
2015-02-03 15:28:36,923 [I/O dispatcher 1] INFO  marytts.R 8 New request (input type "SABLE", output type "AUDIO", voice "cmu-slt-hsmm", effect "", audio "WAVE")
2015-02-03 15:28:36,926 [I/O dispatcher 1] WARN  marytts.server Problem reading input
java.io.FileNotFoundException: http://mary.dfki.de/lib/Sable.v0_2.dtd
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1624)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:637)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startEntity(XMLEntityManager.java:1290)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startDTDEntity(XMLEntityManager.java:1257)
	at com.sun.org.apache.xerces.internal.impl.XMLDTDScannerImpl.setInputSource(XMLDTDScannerImpl.java:263)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.dispatch(XMLDocumentScannerImpl.java:1164)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.next(XMLDocumentScannerImpl.java:1050)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:964)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:606)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:117)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:510)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:848)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:243)
	at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:347)
	at marytts.util.dom.DomUtils.parseDocument(DomUtils.java:142)
	at marytts.util.dom.DomUtils.parseDocument(DomUtils.java:121)
	at marytts.datatypes.MaryData.setData(MaryData.java:221)
	at marytts.server.Request.setInputData(Request.java:245)
	at marytts.server.http.SynthesisRequestHandler.process(SynthesisRequestHandler.java:220)
	at marytts.server.http.SynthesisRequestHandler.handleClientRequest(SynthesisRequestHandler.java:91)
	at marytts.server.http.BaseHttpRequestHandler.handle(BaseHttpRequestHandler.java:138)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler$RequestHandlerAdaptor.handle(BufferingHttpServiceHandler.java:189)
	at org.apache.http.nio.protocol.SimpleNHttpRequestHandler.handle(SimpleNHttpRequestHandler.java:51)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.processRequest(AsyncNHttpServiceHandler.java:453)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.requestReceived(AsyncNHttpServiceHandler.java:225)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler.requestReceived(BufferingHttpServiceHandler.java:127)
	at org.apache.http.impl.nio.DefaultNHttpServerConnection.consumeInput(DefaultNHttpServerConnection.java:161)
	at org.apache.http.impl.nio.DefaultServerIOEventDispatch.inputReady(DefaultServerIOEventDispatch.java:147)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:161)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:335)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:275)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:542)
	at java.lang.Thread.run(Thread.java:745)
2015-02-03 15:28:36,927 [I/O dispatcher 1] INFO  marytts.server Request couldn't be handled successfully.
```

`APML` confirmed:
```
2015-02-03 15:29:15,138 [I/O dispatcher 1] INFO  marytts.R 9 New request (input type "APML", output type "AUDIO", voice "cmu-slt-hsmm", effect "", audio "WAVE")
2015-02-03 15:29:15,141 [I/O dispatcher 1] WARN  marytts.server Problem reading input
java.io.FileNotFoundException: http://mary.dfki.de/lib/apml.dtd
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1624)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:637)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startEntity(XMLEntityManager.java:1290)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startDTDEntity(XMLEntityManager.java:1257)
	at com.sun.org.apache.xerces.internal.impl.XMLDTDScannerImpl.setInputSource(XMLDTDScannerImpl.java:263)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.dispatch(XMLDocumentScannerImpl.java:1164)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.next(XMLDocumentScannerImpl.java:1050)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:938)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:606)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:117)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:510)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:848)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:243)
	at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:347)
	at marytts.util.dom.DomUtils.parseDocument(DomUtils.java:142)
	at marytts.util.dom.DomUtils.parseDocument(DomUtils.java:121)
	at marytts.datatypes.MaryData.setData(MaryData.java:221)
	at marytts.server.Request.setInputData(Request.java:245)
	at marytts.server.http.SynthesisRequestHandler.process(SynthesisRequestHandler.java:220)
	at marytts.server.http.SynthesisRequestHandler.handleClientRequest(SynthesisRequestHandler.java:91)
	at marytts.server.http.BaseHttpRequestHandler.handle(BaseHttpRequestHandler.java:138)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler$RequestHandlerAdaptor.handle(BufferingHttpServiceHandler.java:189)
	at org.apache.http.nio.protocol.SimpleNHttpRequestHandler.handle(SimpleNHttpRequestHandler.java:51)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.processRequest(AsyncNHttpServiceHandler.java:453)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.requestReceived(AsyncNHttpServiceHandler.java:225)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler.requestReceived(BufferingHttpServiceHandler.java:127)
	at org.apache.http.impl.nio.DefaultNHttpServerConnection.consumeInput(DefaultNHttpServerConnection.java:161)
	at org.apache.http.impl.nio.DefaultServerIOEventDispatch.inputReady(DefaultServerIOEventDispatch.java:147)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:161)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:335)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:275)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:542)
	at java.lang.Thread.run(Thread.java:745)
2015-02-03 15:29:15,142 [I/O dispatcher 1] INFO  marytts.server Request couldn't be handled successfully.
```

`ACOUSTPARAMS` confirmed (*ouch!*):
```
2015-02-03 15:31:24,756 [I/O dispatcher 1] INFO  marytts.R 21 New request (input type "ACOUSTPARAMS", output type "AUDIO", voice "cmu-slt-hsmm", effect "", audio "WAVE")
2015-02-03 15:31:24,766 [I/O dispatcher 1] WARN  marytts.server Problem reading input
org.xml.sax.SAXParseException; lineNumber: 7; columnNumber: 47; cvc-complex-type.3.2.2: Attribute 'sampa' is not allowed to appear in element 't'.
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:198)
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.error(ErrorHandlerWrapper.java:134)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:437)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:368)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:325)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator$XSIErrorReporter.reportError(XMLSchemaValidator.java:458)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.reportSchemaError(XMLSchemaValidator.java:3237)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.processAttributes(XMLSchemaValidator.java:2714)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.handleStartElement(XMLSchemaValidator.java:2056)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.startElement(XMLSchemaValidator.java:746)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:378)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2770)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:606)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:117)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:510)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:848)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:243)
	at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:347)
	at marytts.util.dom.DomUtils.parseDocument(DomUtils.java:142)
	at marytts.util.dom.DomUtils.parseDocument(DomUtils.java:121)
	at marytts.datatypes.MaryData.setData(MaryData.java:221)
	at marytts.server.Request.setInputData(Request.java:245)
	at marytts.server.http.SynthesisRequestHandler.process(SynthesisRequestHandler.java:220)
	at marytts.server.http.SynthesisRequestHandler.handleClientRequest(SynthesisRequestHandler.java:91)
	at marytts.server.http.BaseHttpRequestHandler.handle(BaseHttpRequestHandler.java:138)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler$RequestHandlerAdaptor.handle(BufferingHttpServiceHandler.java:189)
	at org.apache.http.nio.protocol.SimpleNHttpRequestHandler.handle(SimpleNHttpRequestHandler.java:51)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.processRequest(AsyncNHttpServiceHandler.java:453)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.requestReceived(AsyncNHttpServiceHandler.java:225)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler.requestReceived(BufferingHttpServiceHandler.java:127)
	at org.apache.http.impl.nio.DefaultNHttpServerConnection.consumeInput(DefaultNHttpServerConnection.java:161)
	at org.apache.http.impl.nio.DefaultServerIOEventDispatch.inputReady(DefaultServerIOEventDispatch.java:147)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:161)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:335)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:275)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:542)
	at java.lang.Thread.run(Thread.java:745)
2015-02-03 15:31:24,767 [I/O dispatcher 1] INFO  marytts.server Request couldn't be handled successfully.
```

All other input types **are working for me**.

**Comment by Munzey** [comment]
I think this is something that could be resolved by first tackling #228

**Comment by Munzey** [comment]
just noticed from your stack trace that you have the output set to `AUDIO`, my issue is regarding the output set to `RAWMARYXML`

**Comment by psibre** [comment]
@Rootex I think we can push this back to `6.0` unless you have a quick fix...

**Comment by Rootex** [comment]
@psibre i have made necessary changes to disable the output that crash for a given input and that works perfectly, however if a new input is selected the output type is not refreshed, one has to manually refresh the page. 

So i have pushed the preliminary changes, but moved to other issues in other to clear my mind and handle the critical ones first.

**Comment by Rootex** [comment]
@attacomsian Here is a link to the JS with what i have written so far. [Javascript for mary web interface](https://github.com/Rootex/marytts/blob/bea516ada53aee88247e55a90adfa6e0695bf315/marytts-runtime/src/main/resources/marytts/server/http/mary.js)

**Comment by attacomsian** [comment]
@psibre I have made required changes to filter the input and output types that crash. There were three input types(`SIMPLEPHONEMES`, `SABLE` & `APML`) for which none of the output types was working. I removed them from input types list. For rest of the input types, I applied filter to only show those output types which never crash.

I have pushed the changes and created new [pull request](https://github.com/marytts/marytts/pull/461).
